### PR TITLE
static analysis framework

### PR DESCRIPTION
**Pull request by yann300**
story https://github.com/ethereum/browser-solidity/issues/252

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -610,6 +611,10 @@ var run = function () {
     }
 
     loadVersion(selectedVersion)
+
+    // static analysis
+    var staticanalysis = new StaticAnalysis(compiler, renderer)
```
Comment (chriseth) [inline src/app.js]: Why do you add it here? It should be rather similar to the "Formal Verification" tab, which is added right at the beginning.

File: src/app/staticanalysis/staticAnalysisView.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  return toRun
+}
+
+staticAnalysisView.prototype.run = function () {
+  if (!this.view) {
+    return
+  }
+  var selected = this.selectedModules()
+  var warningContainer = $('#staticanalysisresult')
+  warningContainer.html('')
```
Comment (chriseth) [inline src/app/staticanalysis/staticAnalysisView.js]: `.empty()`?

File: src/app/staticanalysis/staticAnalysisRunner.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+var AstWalker = require('ethereum-remix').util.AstWalker
+var list = require('./modules/list')
+
+function staticAnalysisRunner () {
+}
+
+staticAnalysisRunner.prototype.run = function (ast, toRun, callback) {
+  var walker = new AstWalker()
+  for (var k in ast) {
+    walker.walk(ast[k].AST, {'*': function (node) {
```
Comment (chriseth) [inline src/app/staticanalysis/staticAnalysisRunner.js]: I think we could improve that in the future. Furthermore, there might be some things to check which need the AST as is, and not just as a callback on each node.

**Comment by yann300** [comment]
TODO 
 - highlight static analysis tab is warning - trigger a run for each compilation)
 - error for each source location
 - nothing if no warning
 - add test

**Pull request merged**